doc/users/installing/index.rst

```diff
@@ -100,76 +100,24 @@ scientific-python-nightly-wheels as the package index to query:
     matplotlib
 
 ======================
-Installing from source
+Install from source
 ======================
 
-If you are interested in contributing to Matplotlib development,
-running the latest source code, or just like to build everything
-yourself, it is not difficult to build Matplotlib from source.
+The following instructions in this section are for very custom installations of
+Matplotlib. Proceed with caution because these instructions may result in your
+build producing unexpected behavior and/or causing local testing to fail.
 
-First you need to install the :ref:`dependencies`.
+.. admonition:: Installing for Development
+  :class: important
 
-A C compiler is required.  Typically, on Linux, you will need ``gcc``, which
-should be installed using your distribution's package manager; on macOS, you
-will need xcode_; on Windows, you will need `Visual Studio`_ 2015 or later.
+  If you would like to contribute to Matplotlib or otherwise would like to
+  install the latest development code, please follow the instructions in
+  :ref:`installing_for_devs`.
```

Comment:
*Sorry, Github doesn't let me do code suggestions on "deleted lines", so only as description:*

- Suggestion: Move the admonition to the top. This is by far the most common case, so let's get it out of the way. Also I think "these instructions may result in your build producing unexpected behavior" does not apply to dev installs.

- Rewirte the section intro (l.106-108) to:

> The following instructions are on installing from source for production use.
This is a very rare case and you should use prebuilt packages instead if possible.
Proceed with caution because these instructions may result in your
build producing unexpected behavior and/or causing local testing to fail.


Possible alternative: Make a dedicated "Install for development" section.